services/ui-src/src/components/report/Elements.tsx

```diff
@@ -88,6 +89,39 @@ export const subHeaderElement = (props: PageElementProps) => {
   );
 };
 
+export const subHeaderMeasureElement = (_props: PageElementProps) => {
+  const { report, currentPageId } = useStore();
+
+  //search for the measure that the currentId is tied to, it maybe a child so we have to check the dependents too.
+  const parentMeasure = report?.measureLookup.defaultMeasures.find(
+    (template) =>
+      template?.measureTemplate === currentPageId ||
+      template?.dependentPages?.find(
+        (page: any) => page.template === currentPageId
+      )
+  );
+
+  //find the parent measure to get the page type
+  const measure = report?.pages.find(
+    (measure) => measure.id === parentMeasure?.measureTemplate
+  ) as MeasurePageTemplate;
```

Comment:
This will deconflict our two branches. I pulled down this branch and tested the changes with optional, required, and substitutions

```suggestion
  //search for the measure that the currentId is tied to, it maybe a child so we have to check the dependents too.
  const measure = report?.pages.find(
    (page) =>
      isMeasurePageTemplate(page) &&
      page.dependentPages &&
      page.dependentPages.find((child) => child.template === currentPageId)
  ) as MeasurePageTemplate;

  return (
    <>
      {measure && (
        <Heading
          as="h2"
          variant="nestedHeading"
          color="#5a5a5a"
          marginBottom="-1.5rem"
        >
          {measure.required ? "Required" : "Optional"} Measure
        </Heading>
      )}
    </>
  );
};
```